### PR TITLE
Add documentation to Abstract Client

### PR DESCRIPTION
**Pull request by cyberhoward**
Adds doc-comments to the abstract-client crate

### CODE REVIEW

File: framework/packages/abstract-client/src/publisher.rs [hunk 1]
```diff
@@ -131,22 +158,29 @@ impl<Chain: CwEnv> Publisher<Chain> {
...
     pub fn account(&self) -> &Account<Chain> {
         &self.account
     }
+
     // TODO: add `account_admin` fn to get the (Sub-)Account's admin.
+    // TODO: I don't get why it's called admin
+    /// Address of the manager
     pub fn admin(&self) -> AbstractClientResult<Addr> {
         self.account.manager()
     }
```
Comment (Buckram123) [inline framework/packages/abstract-client/src/publisher.rs:171]: Should I rename it to manager?

File: framework/packages/abstract-client/src/publisher.rs [hunk 2]
```diff
@@ -115,6 +140,8 @@ impl<Chain: CwEnv> Publisher<Chain> {
             .map_err(Into::into)
     }
 
+    /// Publish Abstract Adapter
+    // TODO: why it's publish app and deploy adapter, shouldn't we stick to one way?
     pub fn deploy_adapter<
```
Comment (Buckram123) [inline framework/packages/abstract-client/src/publisher.rs:145]: Should it be `publish_adapter` or `publish_app` should become `deploy_app`?

File: framework/packages/abstract-client/src/client.rs [hunk 3]
```diff
@@ -9,59 +41,93 @@ use crate::{
...
+    /// # let sender = cosmwasm_std::Addr::unchecked("sender");
+    /// # let chain = cw_orch::prelude::Mock::new(&sender);
+    /// use abstract_client::client::AbstractClient;
+    ///
+    /// let client = AbstractClient::new(chain)?;
+    /// # Ok::<(), AbstractClientError>(())
+    /// ```
     pub fn new(chain: Chain) -> AbstractClientResult<Self> {
+        // TODO: New error type for not found abstract on this chain?
         let abstr = Abstract::load_from(chain)?;
```
Comment (Buckram123) [inline framework/packages/abstract-client/src/client.rs:67]: Loading abstract is completely unchecked, would we prefer to add check on `load_from` or here?

**Review by Buckram123 — COMMENTED** [review]
@CyberHoward Marked few things to discuss while I was documenting it

**Comment by Buckram123** [comment]
@CyberHoward Ready for a review, decided to remove setup macro, because it was one-liner

File: framework/packages/abstract-client/src/account.rs [hunk 4]
```diff
@@ -176,32 +236,43 @@ impl<Chain: CwEnv> Account<Chain> {
...
+    /// This creates a new sub-account and installs the application on it.
+    pub fn install_app<M: ContractInstance<Chain> + InstallConfig + From<Contract<Chain>>>(
         &self,
         configuration: &M::InitMsg,
         funds: &[Coin],
     ) -> AbstractClientResult<Application<Chain, M>> {
         self.install_app_internal(vec![M::install_config(configuration)?], funds)
     }
 
+    pub fn install_adapter<
```
Comment (CyberHoward) [inline framework/packages/abstract-client/src/account.rs:263]: This method doesn't really make sense IMO. 

Adapters should get installed on the current account. Not a new sub-account.

Comment (Buckram123) [reply framework/packages/abstract-client/src/account.rs:263]: Hm, that's an interesting point! Should we have a way to install apps on current account as well?

Comment (CyberHoward) [reply framework/packages/abstract-client/src/account.rs:264]: Or, what about this: 

We add a `install_on_sub_account` boolean attribute to the `Account` that can be set in the builder and defaults to `true`. That way it's a global setting and you can do whatever. 

Want to install adapters on sub-accounts? cool. 
Want to install apps on current account? done.
etc

File: framework/packages/abstract-client/src/application.rs [hunk 5]
```diff
@@ -32,11 +38,15 @@ impl<Chain: CwEnv, M> DerefMut for Application<Chain, M> {
     }
 }
 
-impl<Chain: CwEnv, M> Application<Chain, M> {
-    pub fn new(account: Account<Chain>, module: M) -> Self {
-        Self { account, module }
+impl<Chain: CwEnv, M: RegisteredModule> Application<Chain, M> {
```
[the remark this answers is not in the document]
Comment (Buckram123) [reply framework/packages/abstract-client/src/application.rs:41]: Do you want to revert it!

File: framework/packages/abstract-client/src/account.rs (hunk 4 above)
Comment (Buckram123) [reply framework/packages/abstract-client/src/account.rs:264]: We have multiple ways of obtaining account beyond builder, do we want to have ability to set it on `Account` object as well? And I think depending on a context we might want to use different defaults values
- `from_namespace` -> true
-  `install_*` -> if installed on sub account false(so you don't create another sub-account on top of it accidentally), else true
- `get_last_account` - true

Comment (CyberHoward) [reply framework/packages/abstract-client/src/account.rs:285]: > We have multiple ways of obtaining account beyond builder, do we want to have ability to set it on `Account` object as well? And I think depending on a context we might want to use different defaults values
> 
> * `from_namespace` -> true
> * `install_*` -> if installed on sub account false(so you don't create another sub-account on top of it accidentally), else clone Account object
> * `get_last_account` - true

If we enable fetch_from_namespace by default then we don't need the `from_namespace` anymore.

File: framework/packages/abstract-client/src/application.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (CyberHoward) [reply framework/packages/abstract-client/src/application.rs:41]: I guess not. We'll figure out the standalone case later.

**Review by Buckram123 — APPROVED** [review]

**Pull request merged**